### ISSUE TITLE
Fix code styles of all example solutions

### ISSUE THREAD

**Issue by joelwallis** [exercism/javascript#399]
~~**Blocked by #373**~~

---

We already have ~~an issue to add~~ added linting capabilities to our codebase using eslint. Now we need to make all exercise files compliant with the linting rules. 

We will be spitting it up into exercises and would like to have a separate PR for each exercise.

**Ready to go:**
Below is a list of all files that need linting along with the ones which are already linted (checked).
See the full list of exercises here: https://github.com/exercism/javascript/tree/master/exercises

- [x] accumulate
- [x] acronym
- [x] all-your-base
- [x] allergies
- [x] alphametics
- [x] anagram
- [x] atbash-cipher
- [x] beer-song
- [x] binary
- [x] binary-search
- [x] binary-search-tree
- [x] bob
- [x] bowling
- [x] bracket-push
- [x] change
- [x] circular-buffer
- [x] clock
- [x] collatz-conjecture
- [x] connect
- [x] crypto-square
- [x] custom-set
- [x] diamond
- [x] difference-of-squares
- [x] diffie-hellman
- [x] etl
- [x] flatten-array
- [x] food-chain
- [x] forth
- [x] gigasecond
- [x] grade-school
- [x] grains
- [x] hamming
- [x] hello-world
- [x] hexadecimal
- [x] isbn-verifier
- [x] isogram
- [x] kindergarten-garden
- [x] largest-series-product
- [x] leap
- [x] linked-list
- [x] list-ops
- [x] luhn
- [x] matrix
- [x] meetup
- [x] minesweeper
- [x] nth-prime
- [x] nucleotide-count
- [x] ocr-numbers
- [x] octal
- [x] palindrome-products
- [x] pangram
- [x] pascals-triangle
- [x] perfect-numbers
- [x] phone-number
- [x] pig-latin
- [x] point-mutations
- [x] prime-factors
- [x] protein-translation
- [x] proverb
- [x] pythagorean-triplet
- [x] queen-attack
- [x] raindrops
- [x] reverse-string
- [x] rna-transcription
- [x] robot-name
- [ ] robot-simulator
- [x] roman-numerals
- [x] run-length-encoding
- [x] saddle-points
- [x] say
- [x] scrabble-score
- [x] secret-handshake
- [x] series
- [x] sieve
- [ ] simple-cipher
- [x] simple-linked-list
- [x] space-age
- [x] strain
- [x] sublist
- [x] sum-of-multiples
- [x] transpose
- [x] triangle
- [x] trinary
- [x] twelve-days
- [x] two-bucket
- [x] two-fer
- [x] word-count
- [x] wordy
- [x] zipper

Our CI will check against the linting rules, but since they are not yet implemented for exercises, the [respective directories have been ignored](https://github.com/exercism/javascript/blob/master/.eslintignore).

**Task**
1. Pick one exercise from list of exercises - see if it is not already implemented in the above list (it should not be checked).
2. Comment here that you are working on that exercise.
3. Remove the exercise line from [`.eslintignore`](https://github.com/exercism/javascript/blob/master/.eslintignore)
4. Run `npm run lint` and see all linting errors. 
5. Some of these are auto-fixable using `npm run lint-fix`., others have to be fixed manually.
6. Commit and submit a PR 🎉 

**Note:** Please do not fix `big-integer.js` and `big-integer.spec.js` since those are third party code (for compatibility reasons) and we don't want to touch that. Let it remain in `eslintignore`.

**Comment by Radix16** [comment]
I would like to help with this issue.  It looks like a lot of changes though.  Is this intended to be multiple PRs or just one?   I have installed and executed the linter on the current project and received the linter errors mentioned.

**Comment by rchavarria** [comment]
It will be too much work for one person to fix all those errors. So, if I were you, I would choose an exercise and create a PR that fixes the errors for that single exercise.

**Comment by rchavarria** [comment]
In fact, it seems it's exactly what you're doing. I didn't see those PRs.

**Comment by Radix16** [comment]
The grains exercise has 532 errors reported by the linter.  520 of these errors are all contained in the file big-integer.js which is minified and originates from https://github.com/peterolson/BigInteger.js.  Also the non-minified version of this file does not conform the linter standards set for this project.  I haven't solved this problem myself as a user, but I guess a user is expected to find this library and use it to solve this problem.

How would you like this exercise corrected?   
Would you like me to take the file from https://github.com/peterolson/BigInteger.js and edit it to conform to this project's standard?   
Do you want to leave it as is and live with the linter errors?  
Do you want to find another example solution for this problem that does not use this library?

**Comment by matthewmorgan** [comment]
@Radix16 we should modify the linter settings to skip the vendor files.  If you want to tackle that, it would be appreciated, but I think that's the correct approach.

**Comment by tejasbubane** [comment]
@joelwallis I updated your description to add info about CI and eslintignore I added in https://github.com/exercism/javascript/pull/453. Hope you don't mind :)

**Comment by ajomadlabs** [comment]
@joelwallis I would like to go with word-count exercise

**Comment by tejasbubane** [comment]
Go ahead @ajomadlabs !

**Comment by ajomadlabs** [comment]
@tejasbubane I tested the word-count.spec.js and didn't find any error.

**Comment by ajomadlabs** [comment]
@joelwallis @tejasbubane I have just made a PR regarding the big-integer.spec.js. word-count.spec.js is free of lint errors.

**Comment by tejasbubane** [comment]
If there are no errors in some exercise, please remove it from `.eslintignore` so that we know it's done. And feel free to pick other exercises.

**Comment by matthewmorgan** [comment]
@tejasbubane @ajomadlabs after reviewing that PR, I closed it since we don't really want to make changes in third party libraries.  It's a better practice to leave them in the eslintignore.

See my earlier comment [here](https://github.com/exercism/javascript/issues/399#issuecomment-338434432).

**Comment by ajomadlabs** [comment]
@matthewmorgan Okey. In the above checklist, many files doesn't have any errors like word-count, accumulate etc. I think its high time that we need to update the checklist

**Comment by tejasbubane** [comment]
@ajomadlabs Thanks for your feedback. I have updated the checklist with updated exercise list and check off the ones which don't have any errors.

**Comment by joelwallis** [comment]
@ajomadlabs great work! 🎉

**Comment by minaslater** [comment]
Hi! This is going to be my first contribution in someone else's repo outside of class. O_o
I'm going to try to tackle "perfect numbers".

**Comment by matthewmorgan** [comment]
@minaslater Great!

Please:
1. Fork the repo
2. Make an empty commit
3. Open a PR to this repo with a title like "Dibs - fix lint errors for perfect numbers"
4. Please refer this this issue in the PR description.

Let us know if you need any assistance!

**Comment by slippa91** [comment]
Hi. I will fix queen-attack.

**Comment by minaslater** [comment]
@matthewmorgan Hi! I just saw this reply... I already opened up a PR with the styling fixes. :)

**Comment by joefitz12** [comment]
Wasn't able to push an empty commit, but just submitted a PR for secret-handshake. If it looks good, let me know! Would love to help out some more.

**Comment by natraj-s** [comment]
I'll work on simple-linked-list.

**Comment by cybernetck** [comment]
Hello, I've selected the simple-cipher to work on.  I believe I have the issues resolved, and would like to move forward with a merge request.

**Comment by laurencantlin** [comment]
Fixing nth-prime.

**Comment by matthewmorgan** [comment]
@laurencantlin great!  Please open a PR with an empty commit, and a title like: "Dibs- fix linting issues for nth-prime" 

This will make your intent visible.
